src/dfu.c

```diff
@@ -394,10 +394,10 @@ void dfu_show_info(struct dfu_dev *dfu)
     msg_info("    USB Product         : 0x%04hX\n",
```

Comment:
Shouldn't we also remove the `h` and `(unsigned short)` cast here?

Reply:
No, that would change what happens. Right now printf's %hx will (unsigned short) the promoted int argument that itself already has been (unsigned short)-ed. We can remove either the `h` or the `(unsigned short)` cast of the (assuming at most int-wide) argument and the print will be the same. If we remove both then if the arg doesn't fit in an unsigned short something else will be printed.

Reply:
Here an example:
```
#include <stdio.h>
int main() {
  int i = 0x78000;
  printf("%04hx\n", (unsigned short) i);
  printf("%04x\n", (unsigned short) i);
  printf("%04hx\n", i);
  printf("%04x\n", i);
}
```